js/tji-datasets-explore.js

```diff
@@ -1,15 +1,349 @@
 
+//TODO: make color constant object
+//WRITE documentation for how you expect people to use these classes
 
+var COLOR_TJI_BLUE = '#0B5D93';
+var COLOR_TJI_RED = '#CE2727';
+var COLOR_TJI_DEEPBLUE = '#252939';
+var COLOR_TJI_PURPLE = '#4D3B6B';
+var COLOR_TJI_YELLOW = '#F1AB32';
+var COLOR_TJI_TEAL = '#50E3C2';
+var COLOR_TJI_DEEPRED = '#872729';
+var COLOR_TJI_DEEPPURPLE = '#2D1752';
 
-// MAIN AJAX CALL TO ENDPOINT
 
+// *******************************************************************
+// * "Class" for a single-variable bar chart
+// *
+// * Shows counts of records grouped by a particular column.
+// *
+// * Constructor arguments:
+// *   elt_id: id of HTML canvas element to build Chart on
+// *   groupBy: column to group by
+// *   data: list of objects representing records
+// *   missing_data_label: stand-in label for records missing
+// *                       the groupBy column.
+// *******************************************************************
 
-//TOGGLE CHECKBOXES
-jQuery(document).ready(function($) {
-  $("#filter-section input:checkbox").prop("checked", true);
 
-  $('.js-filter-input-all').click(function() {
-      var $checkboxes = $(this).parent().parent().parent().find('input[type=checkbox]');
-      $checkboxes.prop('checked', $(this).is(':checked'));
-  }); 
-});
\ No newline at end of file
+var TJIGroupByBarChart = function(elt_id, groupBy, data, missing_data_label) {
```

Comment:
Minor cleanup we talked about when we paired that I'm logging here in case someone wants to tackle it. The number of args passed into the constructor is getting gnarly. might make sense to turn this into an objects {elt_id: '', data: data, groupBy:'', ...} so the user doesn't have to know the order of params to get this to work.